sumologic-app-utils/src/sumoresource.py

```diff
@@ -647,38 +717,50 @@ def create_by_install_api(self, appid, appname, source_params, folder_name, *arg
 
-    def update(self, app_folder_id, appname, source_params, appid=None, folder_name=None, retain_old_app=False,
-               s3url=None, *args, **kwargs):
         # Delete is called by CF itself on Old Resource if we create a new resource. So, no need to delete the resource here.
         # self.delete(app_folder_id, remove_on_delete_stack=True)
-        data, new_app_folder_id = self.create(appname, source_params, appid, folder_name, s3url)
+        
+        isAdmin = False
+        if location == "admin":
+            isAdmin=True
+        data, new_app_folder_id = self.create(appname=appname, source_params=source_params, folder_name=folder_name, s3url=s3url, orgID=orgID, share=share, location=location)
         print("updated app appFolderId: %s " % new_app_folder_id)
         if retain_old_app:
-            backup_folder_id = self._create_backup_folder(new_app_folder_id, app_folder_id)
+            backup_folder_id = self._create_backup_folder(new_app_folder_id, app_folder_id, isAdmin)
+            print("backup folder created")
             # Starting Folder Copy
-            response = self.sumologic_cli.copy_folder(app_folder_id, backup_folder_id)
+            response = self.sumologic_cli.copy_folder(app_folder_id, backup_folder_id, isAdmin)
             job_id = response.json()["id"]
             print("Copy Completed parentFolderId: %s jobId: %s" % (backup_folder_id, job_id))
             copied_folder_id = self._wait_for_folder_copy(app_folder_id, job_id)
             # Updating copied folder name with suffix BackUp.
             copied_folder_details = self.sumologic_cli.get_folder_by_id(copied_folder_id)
             copied_folder_details = {"name": copied_folder_details["name"].replace("(Copy)", "- BackUp_" + datetime.now().strftime("%H:%M:%S")),
                                      "description": copied_folder_details["description"][:255]}
-            self.sumologic_cli.update_folder_by_id(copied_folder_id, copied_folder_details)
+            self.sumologic_cli.update_folder_by_id(copied_folder_id, copied_folder_details, isAdmin)
             print("Back Up done for the APP: %s." % backup_folder_id)
         return data, new_app_folder_id
 
-    def delete(self, app_folder_id, remove_on_delete_stack, *args, **kwargs):
+    def delete(self, app_folder_id, remove_on_delete_stack, location=None, *args, **kwargs):
+        isAdmin = False
+        if location == "admin":
+            isAdmin=True
         if remove_on_delete_stack:
-            response = self.sumologic_cli.delete_folder(app_folder_id)
-            print("deleting app folder %s : %s" % (app_folder_id, response.text))
+            try:
+                response = self.sumologic_cli.delete_folder(app_folder_id, isAdmin)
+                print("deleting app folder %s : %s" % (app_folder_id, response.text))
+            except Exception as e:
+                print("App - Exception while deleting the App folder %s," % e)
```

Comment:
Update print("App - Exception while deleting the App folder %s," % e) to print("App - Exception while deleting the App folder ID %s, error: %s " %(app_folder_id, e)